### PR TITLE
Prepare for release

### PR DESCRIPTION
**Pull request by hfcorreia**
- [x] Add readme documentation
- [x] Add godoc

### CODE REVIEW

**Review by raffo0707 — APPROVED** [review]

File: dynamodb.go [hunk 1]
```diff
@@ -20,13 +20,41 @@ const (
...
-// DynamoDBAPI just a wrapper over aws-sdk dynamodbiface.DynamoDBAPI interface for mocking purposes
-type DynamoDBAPI interface {
+// DynamoDBClient is a wrapper interface over aws-sdk dynamodbiface.DynamoDBClient for mocking purposes
+type DynamoDBClient interface {
 	dynamodbiface.DynamoDBAPI
 }
 
+// NewDynamoClient creates a new DynamoDBClient to create a session for a given ARN role
+//
+// NewDynamoClient creates DynamoDB client instance a wrapper around the AWS-SDK with a predefined Session.
```
Comment (jmdalmeida) [inline dynamodb.go]:
```suggestion
// NewDynamoClient creates a DynamoDB client wrapper around the AWS-SDK with a predefined Session.
```

File: dynamodb.go [hunk 2]
```diff
@@ -20,13 +20,41 @@ const (
...
+// By default, it creates a new Session with SharedConfigEnable,
+// so you can use AWS SDK's environment variables and AWS credentials to connect to DynamoDB.
+//
+// The provided ARN role allows you to configure the Session to ito assume a specific IAM Role.
```
Comment (jmdalmeida) [inline dynamodb.go]:
```suggestion
// The provided ARN role allows you to configure the Session to assume a specific IAM Role.
```

File: dynamodb.go [hunk 3]
```diff
@@ -20,13 +20,41 @@ const (
...
 )
 
-// DynamoDBAPI just a wrapper over aws-sdk dynamodbiface.DynamoDBAPI interface for mocking purposes
-type DynamoDBAPI interface {
+// DynamoDBClient is a wrapper interface over aws-sdk dynamodbiface.DynamoDBClient for mocking purposes
+type DynamoDBClient interface {
 	dynamodbiface.DynamoDBAPI
 }
 
+// NewDynamoClient creates a new DynamoDBClient to create a session for a given ARN role
```
Comment (jmdalmeida) [inline dynamodb.go]: remove this as it is the same info as line 34

File: dynamodb.go [hunk 4]
```diff
@@ -95,14 +111,21 @@ func (db dynamoDBSerivce) UpdateCapacity(capacity Capacity) error {
...
-	_, err := db.api.UpdateTable(input)
+	_, err := db.client.UpdateTable(input)
 	if err != nil {
 		return fmt.Errorf("unable to update table %s: %s", db.tableName, err)
 	}
 
 	return db.WaitForReadyTable()
 }
 
+// BatchWrite writes given DynamoDBItem in the DynamoDB table.
```
Comment (jmdalmeida) [inline dynamodb.go]:
```suggestion
// BatchWrite writes the given DynamoDBItem slice into the DynamoDB table.
```

File: dynamodb.go [hunk 5]
```diff
@@ -95,14 +111,21 @@ func (db dynamoDBSerivce) UpdateCapacity(capacity Capacity) error {
...
+//
+// BatchWrite splits the given DynamoDBItem slice in 25 of 25 items.
```
Comment (jmdalmeida) [inline dynamodb.go]:
```suggestion
// The given items will be written in groups of 25 each.
```

File: dynamodb.go [hunk 6]
```diff
@@ -207,6 +231,8 @@ func (db dynamoDBSerivce) retry(handler func(attempt, elapsed int) (bool, error)
 	return fmt.Errorf("waited for too long (%d ms) to perform operation on %s table", elapsed, db.tableName)
 }
 
+// Scan allows you to perform a parallel scan over the table, writing the scanned items into the provided itemsChan
+// If totalSegments is equal to 1, it will a non-parallel scan.
```
Comment (jmdalmeida) [inline dynamodb.go]:
```suggestion
// If totalSegments is equal to 1, it will perform a sequential scan.
```

File: sleeper.go [hunk 7]
```diff
@@ -5,15 +5,16 @@ import (
...
 
-const (
-	maxRandomFactor = 100
-)
+const maxRandomFactor = 100
 
-// Sleeper abstracts out sleep side effects to allow better testing
-type Sleeper func(elapsedMilliseconds int) int
+// Sleeper abstracts out sleep side effects to allow for better testing,
+// receiving the ms it to sleep, returning the total time it spent sleeping
```
Comment (jmdalmeida) [inline sleeper.go]:
```suggestion
// receiving the intended sleep time (in ms) and returning the total time spent sleeping
```

File: sleeper.go [hunk 8]
```diff
@@ -5,15 +5,16 @@ import (
...
+type Sleeper func(ms int) int
 
-func RandomSleeper(elapsedMilliseconds int) int {
-	elapsed := elapsedMilliseconds + rand.Intn(maxRandomFactor)
+// RandomSleeper will sleep for the provided ms plus an aditional factor (ranging between [0, 100[).
```
Comment (jmdalmeida) [inline sleeper.go]:
```suggestion
// RandomSleeper will sleep for the provided time (in ms) plus an additional random factor (ranging between [0, 100[).
```

**Review by jmdalmeida — CHANGES_REQUESTED** [review]
Left some suggestions, great job! 💪

**Review by jmdalmeida — APPROVED** [review]

**Pull request merged**